include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -2439,7 +2440,7 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::reinit(
   AssertThrow(this->mapping_info_on_the_fly.get() != nullptr,
               ExcNotImplemented());
 
-  this->mapping_info->reinit(cell, unit_points);
+  this->mapping_info_on_the_fly->reinit(cell, unit_points);
```

Comment:
So, was this a bug before?

Reply:
No, it wasn't. For this `reinit` function this constructor needs to be used.
https://github.com/dealii/dealii/blob/8cf211aa2568fd3eab2be1c35543b5121b0a3103/include/deal.II/matrix_free/fe_point_evaluation.h#L1840-L1844
So, `mapping_info` and `mapping_info_on_the_fly` point to the same object.